basicsr/archs/hifacegan_arch.py

```diff
@@ -0,0 +1,265 @@
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+# 20210519 [Lotayou]: Currently Pytorch official SyncBatchNorm only works under DDP training.
+# from basicsr.models.archs.sync_batchnorm import SynchronizedBatchNorm2d
+from basicsr.utils.registry import ARCH_REGISTRY
+from .hifacegan_util import BaseNetwork, LIPEncoder, SPADEResnetBlock, get_nonspade_norm_layer
+
+
+@ARCH_REGISTRY.register()
+class SPADEGenerator(BaseNetwork):
+    """Generator with SPADEResBlock"""
+
+    def __init__(self,
+                 num_in_ch=3,
+                 num_feat=64,
+                 use_vae=False,
+                 z_dim=256,
+                 crop_size=512,
+                 norm_g='spectralspadesyncbatch3x3',
+                 is_train=True,
+                 init_train_phase=3):  # progressive training disabled
+        super().__init__()
+        self.nf = num_feat
+        self.input_nc = num_in_ch
+        self.is_train = is_train
+        self.train_phase = init_train_phase
+
+        self.scale_ratio = 5  # hardcoded now
+        self.sw = crop_size // (2**self.scale_ratio)
+        self.sh = self.sw  # 20210519: By default use square image, aspect_ratio = 1.0
```

Comment:
I think the date is not necessary, because git will track the date

Reply:
Got it, will search and delete all date comments